src/pixelator/graph/backends/implementations.py

```diff
@@ -428,9 +517,104 @@ def __init__(
         """
         self._raw = raw
 
+    @staticmethod
+    def _build_plain_graph_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: Union[nx.Graph, nx.MultiGraph],
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        g = nx.empty_graph(0, create_using)
+        g.add_edges_from(
+            (row[0], row[1])
+            for row in df.collect(streaming=True).iter_rows(
+                named=False, buffer_size=1000
+            )
+        )
+        return g
+
+    @staticmethod
+    def _build_graph_with_node_counts_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: bool,
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        unique_markers = set(df.unique("marker").collect()["marker"].to_list())
+        initial_marker_dict = {marker: 0 for marker in unique_markers}
+
+        def _add_or_append_to_node(g, node, marker):
+            if g.nodes.get(node):
+                g.nodes.get(node)["markers"][marker] += 1
```

Comment:
Maybe save the extra lookup here since this is quite performance sensitive?
```suggestion
            existing_node = g.nodes.get(node)
            if existing_node:
                existing_node["markers"][marker] += 1
```

Reply:
Good call!